cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java

```diff
@@ -199,9 +199,6 @@ private void updateIndex(final String indexName,
                                                             initialArgs,
                                                             indexName,
                                                             rev);
-                if (parameter == null) {
```

Comment:
Correct, populateDBParameter would put a `null` into `contentValues` if the type wasn't one we understood (https://github.com/cloudant/sync-android/blob/master/cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexUpdater.java#L330). And then that `null` would be skipped later, all entirely pointless.

However, that's for one `contentValues` entry on `DBParameter`.  These null checks are correct to be removed since `populateDBParameter` never returns null.